runtime/compiler/optimizer/InterpreterEmulator.cpp

```diff
@@ -603,13 +606,40 @@ InterpreterEmulator::visitInvokedynamic()
    TR::Node *callNode = 0;
    TR::ResolvedMethodSymbol *resolvedSymbol = 0;
    Operand *result = NULL;
-
-   TR_ResolvedMethod * owningMethod = _methodSymbol->getResolvedMethod();
+   int32_t callSiteIndex = next2Bytes();
    TR::KnownObjectTable *knot = comp()->getOrCreateKnownObjectTable();
-   if (knot && !owningMethod->isUnresolvedCallSiteTableEntry(cpIndex))
+   TR_ResolvedJ9Method * owningMethod = static_cast<TR_ResolvedJ9Method*>(_methodSymbol->getResolvedMethod());
+
+#if defined(J9VM_OPT_OPENJDK_METHODHANDLE)
+   if (!knot && owningMethod->isUnresolvedCallSiteTableEntry(callSiteIndex)) return;
```

Comment:
Should this be `||`? Also, I don't think we need `knot` to inline the adapter method. Adapter method is static and is known if the callsite is resolved.